src/OrleansSQLUtils/CreateOrleansTables_PostgreSql.sql

```diff
@@ -0,0 +1,937 @@
+        @Generation,
+        @SiloName,
+        @HostName,
+        @Status,
+        @ProxyPort,
+        @StartTime,
+        @IAmAliveTime,
+        @Version
+    );
+');
+
+
+
+
+
+
+
+CREATE FUNCTION update_membership(
+    DeploymentIdArg OrleansMembershipTable.DeploymentId%TYPE, 
+    AddressArg      OrleansMembershipTable.Address%TYPE, 
+    PortArg         OrleansMembershipTable.Port%TYPE, 
+    GenerationArg   OrleansMembershipTable.Generation%TYPE,
+    StatusArg       OrleansMembershipTable.Status%TYPE,
+    SuspectTimesArg OrleansMembershipTable.SuspectTimes%TYPE,
+    IAmAliveTimeArg OrleansMembershipTable.IAmAliveTime%TYPE,
+    VersionArg      OrleansMembershipVersionTable.Version%TYPE
+  )
+  RETURNS TABLE(row_count integer) AS
+$func$
+DECLARE
+    RowCountVar int := 0;
+BEGIN
+  
+  
+    BEGIN
+      
+	UPDATE OrleansMembershipVersionTable
+	SET
+		Timestamp = (now() at time zone 'utc'),
+		Version = Version + 1
+	WHERE
+		DeploymentId = DeploymentIdArg AND DeploymentIdArg IS NOT NULL
+		AND Version = VersionArg AND VersionArg IS NOT NULL;
+
+        
+    GET DIAGNOSTICS RowCountVar = ROW_COUNT;
+        
+	UPDATE OrleansMembershipTable
+	SET
+		Status = StatusArg,
+		SuspectTimes = SuspectTimesArg,
+		IAmAliveTime = IAmAliveTimeArg
+	WHERE
+		DeploymentId = DeploymentIdArg AND DeploymentIdArg IS NOT NULL
+		AND Address = AddressArg AND AddressArg IS NOT NULL
+		AND Port = PortArg AND PortArg IS NOT NULL
+		AND Generation = GenerationArg AND GenerationArg IS NOT NULL
+		AND RowCountVar > 0;
+
+
+        GET DIAGNOSTICS RowCountVar = ROW_COUNT;
+
+        ASSERT RowCountVar <> 0, 'no rows affected, rollback';
+
+        
+        RETURN QUERY SELECT RowCountVar;
+    EXCEPTION
+    WHEN assert_failure THEN
+        RETURN QUERY SELECT RowCountVar;
+    END;
+    
+END
+$func$ LANGUAGE plpgsql;
+
+
+INSERT INTO OrleansQuery(QueryKey, QueryText)
+VALUES
+(
+	'UpdateMembershipKey','
+    SELECT * FROM update_membership(    
+        @DeploymentId, 
+        @Address, 
+        @Port, 
+        @Generation,
+        @Status,
+        @SuspectTimes,
+        @IAmAliveTime,
+        @Version
+    );
+');
+
+
+
+
+
+
+
+CREATE FUNCTION upsert_reminder_row(
+    ServiceIdArg    OrleansRemindersTable.ServiceId%TYPE, 
+    GrainIdArg      OrleansRemindersTable.GrainId%TYPE, 
+    ReminderNameArg OrleansRemindersTable.ReminderName%TYPE,     
+    StartTimeArg    OrleansRemindersTable.StartTime%TYPE,
+    PeriodArg       OrleansRemindersTable.Period%TYPE,
+    GrainHashArg    OrleansRemindersTable.GrainHash%TYPE
+  )
+  RETURNS TABLE(version integer) AS
+$func$
+DECLARE
+    VersionVar int := 0;  
+BEGIN
+  
+    
+  
+    
+	INSERT INTO OrleansRemindersTable
+	(
+		ServiceId,
+		GrainId,
+		ReminderName,
+		StartTime,
+		Period,
+		GrainHash,
+		Version
+	)
+	SELECT
+		ServiceIdArg,
+		GrainIdArg,
+		ReminderNameArg,
+		StartTimeArg,
+		PeriodArg,
+		GrainHashArg,
+		0
+    ON CONFLICT (ServiceId, GrainId, ReminderName) 
```

Comment:
Yes -- the upsert statement is highly consistent too -- two statements running at the same time will never error -- unlike TSQL's Merge statement.  But then again, TSQL's Merge has more features.